frontend/dockerfile/docs/reference.md

```diff
@@ -803,17 +803,21 @@ This mount type allows mounting `tmpfs` in the build container.
 
 ### RUN --mount=type=secret
 
-This mount type allows the build container to access secure files such as
-private keys without baking them into the image.
-
-| Option                         | Description                                                                                       |
-| ----------                     | ------------------------------------------------------------------------------------------------- |
-| `id`                           | ID of the secret. Defaults to basename of the target path.                                        |
-| `target`, `dst`, `destination` | Mount path. Defaults to `/run/secrets/` + `id`.                                                   |
-| `required`                     | If set to `true`, the instruction errors out when the secret is unavailable. Defaults to `false`. |
-| `mode`                         | File mode for secret file in octal. Default `0400`.                                               |
-| `uid`                          | User ID for secret file. Default `0`.                                                             |
-| `gid`                          | Group ID for secret file. Default `0`.                                                            |
+This mount type allows the build container to access secret values, such as
+tokens or private keys, without baking them into the image.
+
+By default, the secret is mounted as a file. You can also mount the secret as
+an environment variable by setting the `env` option.
+
+| Option                         | Description                                                                                                     |
+| ------------------------------ | --------------------------------------------------------------------------------------------------------------- |
+| `id`                           | ID of the secret. Defaults to basename of the target path.                                                      |
+| `target`, `dst`, `destination` | Mount the secret to the specified path. Defaults to `/run/secrets/` + `id` if unset and if `env` is also unset. |
+| `env`                          | Mount the secret to an environment variable instead of a file, or both. (since Dockerfile v1.10.0)              |
```

Comment:
Should we have a column with minimal version similar to https://github.com/moby/buildkit/pull/5275 ?

Reply:
yeah thought about it. wdyt? Do you know when those other fields were added? :D

Reply:
Looking at https://github.com/moby/buildkit/blame/0af7b1b9c693cf3c4be62d80beb3fd5b218a12f9/frontend/dockerfile/docs/experimental.md, other attributes are there since 1.0.

Reply:
OK maybe a table will look silly in that case. That's kinda why I didn't do it immediately. I didn't know when any of the other opts were released, and having them all empty or say "1.0" is a bit weird.

Reply:
Yes agree let's keep it like this for now